backend/functions/views.py

```diff
@@ -1,30 +1,145 @@
 from rest_framework.response import Response
-from dataset import models as dataset_models
-from projects.models import *
 from tasks.models import *
-import json
+from users.utils import LANG_NAME_TO_CODE_ULCA, LANG_TRANS_MODEL_CODES
+
+## Utility functions
+def get_translation_using_cdac_model(input_sentence, source_language, target_language):
+    """Function to get the translation for the input sentences using the CDAC model.
+
+    Args:
+        input_sentence (str): Sentence to be translated.
+        source_language (str): Original language of the sentence.
+        target_language (str): Final language of the sentence.
+
+    Returns:
+        str: Translated sentence.
+    """
+
+    # Get the translation model ID
+    model_id = LANG_TRANS_MODEL_CODES.get(f"{source_language}-{target_language}", 144)
+
+    # Convert language names to the language code
+    source_language = LANG_NAME_TO_CODE_ULCA[source_language]
+    target_language = LANG_NAME_TO_CODE_ULCA[target_language]
+
+    headers = {
+        # Already added when you pass json= but not when you pass data=
+        # 'Content-Type': 'application/json',
+    }
+
+    json_data = {
+        "input": [
+            {
+                "source": input_sentence,
+            },
+        ],
+        "config": {
+            "modelId": model_id,
+            "language": {
+                "sourceLanguage": source_language,
+                "targetLanguage": target_language,
+            },
+        },
+    }
+
+    response = requests.post(
+        "https://cdac.ulcacontrib.org/aai4b-nmt-inference/v0/translate",
+        headers=headers,
+        json=json_data,
+    )
+
+    return response.json()["output"][0]["target"]
 
-@api_view(['POST'])
+
+def save_translation_pairs(
+    languages, input_sentences, output_dataset_instance_id, input_dataset_instance_id
+):
+    """Function to save the translation pairs in the database.
+
+    Args:
+        languages (list): List of output languages for the translations.
+        input_sentences (list(list)): List of input sentences in list format containing the - [input_sentence, input_language, context, quality_status]
+        output_dataset_instance_id (int): ID of the output dataset instance.
+        input_dataset_instance_id (int): ID of the input dataset instance.
+    """
+
+    # Iterate through the languages
+    for output_language in languages:
+
+        # Save all the translation outputs in the TranslationPair object
+        for (
+            sentence_text_id,
+            input_sentence,
+            language,
+            context,
+            quality_status,
+            metadata,
+        ) in input_sentences:
+
+            # Only perform the saving if quality status is clean and corrected text is not empty
+            if (quality_status == "Clean") and (input_sentence is not None):
+
+                # Get the translations
+                translated_sentence = get_translation_using_cdac_model(
```

Comment:
One request:

I think translating line-by-line will be quite time-consuming.
Can you please check once with Aravinth if there's an CDAC IndicTrans endpoint that supports batch translation of multiple sentences?

If that's supported, can you please ask him the optimal batch_size and do a batch-inference instead of sentence-by-sentence?

Reply:
We can discuss with @aravinth here directly about the batch translations in IndicTrans models.

Reply:
Hi @prakharrathi25 @GokulNC , 
Batching is supported in NMT. (input object is an array)
Ideal batch size = 50 to 75

<img width="1011" alt="image" src="https://user-images.githubusercontent.com/1707796/182310470-d277fb9b-909f-455f-a5a3-18ab8c409250.png">

Reply:
Thanks, I will look at this.

Reply:
Hi, @GokulNC, I have added batch translations to the `functions`. Please let me know if more changes are required.